xDiffCollection/Classes/CollectionBin.swift

```diff
@@ -81,6 +90,29 @@ public struct CollectionBin<T, Backstorage: RangeReplaceableCollection>: Collect
         
         return updating(with: _backstorage.replacing(with: element, at: index ))
     }
+    
+    //Linear insert function assumes that optional _sort is NOT nil
+    private func linearInsert(_ element:T) -> Backstorage {
+        var inserted = false
+        
+        var resp = Backstorage()
+        //Do a linear search for the proper place for insertation:
+        for e in _backstorage {
```

Comment:
Pretty smart approach. Hard to understand during the first glance how it supports two way of sorting though :)